pkg/amazon/backend/convert.go

```diff
@@ -37,50 +39,118 @@ func Convert(project *types.Project, service types.ServiceConfig) (*ecs.TaskDefi
 			fmt.Sprintf(" %s.local", project.Name),
 		}))
 
-	return &ecs.TaskDefinition{
-		ContainerDefinitions: []ecs.TaskDefinition_ContainerDefinition{
-			{
-				Command:               service.Command,
-				DisableNetworking:     service.NetworkMode == "none",
-				DnsSearchDomains:      service.DNSSearch,
-				DnsServers:            service.DNS,
-				DockerSecurityOptions: service.SecurityOpt,
-				EntryPoint:            service.Entrypoint,
-				Environment:           toKeyValuePair(service.Environment),
-				Essential:             true,
-				ExtraHosts:            toHostEntryPtr(service.ExtraHosts),
-				FirelensConfiguration: nil,
-				HealthCheck:           toHealthCheck(service.HealthCheck),
-				Hostname:              service.Hostname,
-				Image:                 service.Image,
-				Interactive:           false,
-				Links:                 nil,
-				LinuxParameters:       toLinuxParameters(service),
-				LogConfiguration: &ecs.TaskDefinition_LogConfiguration{
-					LogDriver: ecsapi.LogDriverAwslogs,
-					Options: map[string]string{
-						"awslogs-region":        cloudformation.Ref("AWS::Region"),
-						"awslogs-group":         cloudformation.Ref("LogGroup"),
-						"awslogs-stream-prefix": project.Name,
-					},
+	logConfiguration := &ecs.TaskDefinition_LogConfiguration{
+		LogDriver: ecsapi.LogDriverAwslogs,
+		Options: map[string]string{
+			"awslogs-region":        cloudformation.Ref("AWS::Region"),
+			"awslogs-group":         cloudformation.Ref("LogGroup"),
+			"awslogs-stream-prefix": project.Name,
+		},
+	}
+
+	var (
+		containers     []ecs.TaskDefinition_ContainerDefinition
+		volumes        []ecs.TaskDefinition_Volume
+		mounts         []ecs.TaskDefinition_MountPoint
+		initContainers []ecs.TaskDefinition_ContainerDependency
+	)
+	if len(service.Secrets) > 0 {
+		volumes = append(volumes, ecs.TaskDefinition_Volume{
+			Name: "secrets",
+		})
+		mounts = append(mounts, ecs.TaskDefinition_MountPoint{
```

Comment:
Is there a way to use `memfs` for the secrets? IIRC, Swarm services used `memfs` to prevent the secrets from getting stored in the filesystem (but I'd have to dig how it's set up exactly)

Reply:
not supported on Fargate